3-UsingAPIs/Week1/homework/ex1-johnWho.js

```
@@ -8,22 +9,20 @@ Rewrite this function, but replace the callback syntax with the Promise syntax:
 - If the Promise `rejects`, pass an error as the argument to reject with: "You 
   didn't pass in a first name!"
 ------------------------------------------------------------------------------*/
-// TODO see above
-const getAnonName = (firstName, callback) => {
-  setTimeout(() => {
-    if (!firstName) {
-      callback(new Error("You didn't pass in a first name!"));
-      return;
+const getAnonName = (firstName) => {
+  return new Promise((resolve, reject) => {
+    if (typeof firstName === 'string' && arguments.length > 0) {
+      resolve(`${firstName} Doe`);
+    } else {
+      reject(new Error("You didn't pass in a first name!"));
     }
-
-    const fullName = `${firstName} Doe`;
-
-    callback(fullName);
-  }, 1000);
+  });
 };
 
 function main() {
-  getAnonName('John', console.log);
+  getAnonName()
+    .then((message) => console.log(message))
+    .catch(() => new Error());
```

In this case, you won't see the error message that you added up in line 17! so you don't need to return a new error here

---

i changed it 👍